### PR TITLE
412b csl license

### PR DESCRIPTION
**Pull request by robmoffat**

### CODE REVIEW

**Comment by robmoffat** [comment]
ok, changed the CONTRIBUTING, README and GOVERNANCE files.

File: GOVERNANCE.md
Comment (maoo) [inline GOVERNANCE.md:17]: @robmoffat - it seems that "Voting Participants" are different than "Participants", whereas my understanding is that all Participants have the right to vote.

By reading the entire file, I think that what you write makes sense, but we're effectively adding a new Role, which IMO needs some review and approval from the FDC3 (core) maintainer's team.

File: README.md
Comment (maoo) [inline README.md:92]: @robmoffat is that `FDC3 x` a typo? Shouldn't it be `FDC3 Maintainers` ?

File: README.md
Comment (maoo) [inline README.md:129]: @robmoffat - wrong link here, please replace `[Contributing](CONTRIBUTING)` with `[Contributing](CONTRIBUTING.md)`

File: README.md
Comment (maoo) [inline README.md:144]: @robmoffat - you can also use `[Governance document](GOVERNANCE.md#1roles)

File: README.md
Comment (maoo) [inline README.md:148]: @robmoffat - why using the term "Contributors" here? Shouldn't we stick to the roles defined in Governance.md , to avoid confusion?

**Review by maoo — CHANGES_REQUESTED** [review]
@robmoffat - great work here, I've added my comments below.

File: GOVERNANCE.md
Comment (robmoffat) [reply GOVERNANCE.md:17]: I think we should check with @kriswest, but really this is just clarifying what was already the case I think

File: README.md
Comment (robmoffat) [reply README.md:92]: yes, probably me trying to cut something -good catch

File: README.md
Comment (robmoffat) [reply README.md:148]: This is a good question and I'm interested to see what you think about it.   So, we don't define what a participant is in the README file, so I didn't feel it was the right place to introduce that term.  Also, the section is called "Contributing" in the README... so it felt more natural to continue to call it that.  However, if we want to avoid the noun "Contributor" we could say something like:

> When contributing to FINOS standards projects, you should follow... 

Would that be better do you think?

Comment (maoo) [reply README.md:148]: Yes, I think your suggestion makes sense.

I would also suggest to replace `Please see the Contributing guide for details on how to contribute to FDC3.` with

`Please see the Governance and Contributing guides for details on how to contribute to FDC3`, as it would allow the reader to navigate to the roles definitions. WDYT?

**Review by maoo — APPROVED** [review]
LGTM!

File: GOVERNANCE.md
Comment (kriswest) [inline GOVERNANCE.md:31]: Needs to refer back to the SWG for decisions

```suggestion
The Working Group may form one or more "Discussion Groups" to organize collaboration around a particular aspect of a specification. Discussion Groups are for discussion only.  Approval of all portions of a specification is subject to the consensus-based decision-making process of the FDC3 Standard Working Group.
```

File: GOVERNANCE.md
Comment (kriswest) [reply GOVERNANCE.md:17]: Rob is correct that it's always been like this - they are not intended to be separate groups, but not every participant gets around to signing up for voting and agreeing to the terms. The sign-up has always been a way to get people to agree to the license terms and get that documented - its not a great process, but with participants that can't do Github, we haven't had a reasonable alternative proposed.

To deal with @maoo's concern we could just tweak the structure here to keep them together. Will propose an edit

File: GOVERNANCE.md
Comment (kriswest) [inline GOVERNANCE.md:27]:
```suggestion
#### Register to vote
Participants may register to vote on changes to the FDC3 Standard (see [Section 2](#2decision-making) below). To do so use this link: [fdc3-participants+subscribe@finos.org](mailto:fdc3-participants+subscribe@finos.org?subject=Please%20enroll%20me%20as%20FDC3%20standard%20participant&amp;cc=fdc3-pmc%40finos.org&amp;body=HI%2C%20my%20name%20is%20%3CFirstName%20LastName%3E%20and%20I%27d%20like%20to%20formally%20participate%20to%20the%20FDC3%20standard%20process.%20I%20plan%20to%20contribute%20as%20%3Cindividual%7Con%20behalf%20of%20organizationName%3E%20and%20I%20have%20reviewed%20the%20policies%20described%20at%20https%3A%2F%2Fgithub.com%2Ffinos%2Fcommunity%2Ftree%2Fmaster%2Fgovernance%2FStandards-Projects%20.%20Thank%20you!) to send a templated email email to join the enrolled voting participants group.

Upon enrollment as an [FDC3 voting participant](https://github.com/orgs/finos/teams/fdc3-participants), you will be invited to join the [FINOS GitHub organization](https://github.com/orgs/finos/people) and the [fdc3-participants](https://github.com/orgs/finos/teams/fdc3-participants) GitHub team.

- [View the FDC3 Participants GitHub Team](https://github.com/orgs/finos/teams/fdc3-participants)
```

File: GOVERNANCE.md
Comment (kriswest) [inline GOVERNANCE.md:29]: Renumber after merging voting participants and participants
```suggestion
### 1.2. Discussion Groups
```

File: GOVERNANCE.md
Comment (kriswest) [inline GOVERNANCE.md:33]: Renumber after merging voting participants and participants
```suggestion
### 1.3.	 Maintainers
```

File: GOVERNANCE.md
Comment (kriswest) [inline GOVERNANCE.md:39]: Renumber after merging voting participants and participants
```suggestion
### 1.4. Editors
```

File: GOVERNANCE.md
Comment (kriswest) [inline GOVERNANCE.md:45]: Shouldn't be a subheading under editors
```suggestion
### How do you become an Editor or Maintainer?
```

File: LICENSE
Comment (kriswest) [inline LICENSE:51]: Is this default correct or should it be Apache?

File: PATENTS.md
Comment (kriswest) [inline PATENTS.md:1]: There is also a version of this in the charter document - however, I think we need it somewhere in the root of the project - either in the LICENSE file or NOTICES.md file (tack it onto the end after the license acceptance section for the CSL), particularly as these license terms still govern past contributions from participants who have not yet accepted the CSL.

I am aware that PATENTS-FDC3-1.0.md is not removed in this PR - but it's not covering 1.2-2.0.

File: README.md
Comment (kriswest) [inline README.md:156]:
```suggestion
- See [Licenses in the FDC3 Charter](https://fdc3.finos.org/docs/fdc3-charter) for details of licensing for previous versions of the FDC3 Standard and [NOTICES.md](./NOTICES.md) for details of license acceptance, withdrawals and exclusions. 
```

File: docs/fdc3-charter.md
Comment (kriswest) [inline docs/fdc3-charter.md:50]: Include future adopted versions as well as drafts
```suggestion
 - Versions 2.1 and subsequent FDC3 specifications and draft specifications are licensed under the [Community Specification License 1.0](https://github.com/finos/FDC3/blob/45ca765e5ff9a44a1fa5437eb70cf876bf4898aa/LICENSE)
```

File: GOVERNANCE.md
Comment (kriswest) [inline GOVERNANCE.md:25]: The email link/template here needs to be updated to agree to the CSL license files in this repo rather than the FINOS community one that it links to.

**Review by kriswest — CHANGES_REQUESTED** [review]
Mostly good, I've suggested a few tweaks that I think need to be made. In particular, I think past licenses should be somewhere in the root in addition to on the FDC3 Charter page and we can combine the Participants and Voting participants to deal with @maoo's concern on that one (we don't want people thinking of voters as a different group - rather signing up to vote is a step all participants should take).

File: LICENSE
Comment (maoo) [reply LICENSE:51]: This is correct, and copied from https://github.com/CommunitySpecification/Community_Specification/blob/main/1._Community_Specification_License-v1.md

The link with Apache is when it says `If no source code license is included, the source code will be subject to the MIT License.` , and that's why we have added mentions to Apache in the README.

If you think we should make this more prominent, we could add a `SOFTWARE-LICENSE` in the root folder, containing Apache 2.0 License.

File: CONTRIBUTING.md
Comment (maoo) [reply CONTRIBUTING.md:102]: Good catch.

If you need a README for Docusaurus, you can start from https://github.com/finos/software-project-blueprint/tree/main/website

I also realised that you don't have the `Edit page` feature enabled, as we do in https://community.finos.org/docs/journey/learn (check the bottom of the page), which may definitely help people submitting changes.

File: docs/fdc3-charter.md
Comment (kriswest) [inline docs/fdc3-charter.md:50]: @robmoffat @maoo it just occured to me (while writing the FDC3 2.1 adoption email) that It's actually version 1.1 - **2.1** that are under the FINOS IP policy. we're making this change straight after 2.1 so its all versions subsequent to that that are under CSL!

File: CONTRIBUTING.md
Comment (robmoffat) [reply CONTRIBUTING.md:103]: I think this link should just go to fdc3.finos.org for now

File: GOVERNANCE.md
Comment (robmoffat) [reply GOVERNANCE.md]: So, a link to GOVERNANCE.md instead?

File: docs/fdc3-charter.md
Comment (robmoffat) [reply docs/fdc3-charter.md]: ok, updated.

File: README.md
Comment (robmoffat) [reply README.md:152]: @maoo this underlines the fact that we have to merge this PR **after** the 2.1 release is cut, and when we move docusaurus to version "next".  

At that point, this version of the FDC3 charter will be correct and we can fix the README at the same time.

File: CONTRIBUTING.md
Comment (kriswest) [reply CONTRIBUTING.md:103]: yep, thats fine @robmoffat, unless or until we write some guidance for future maintainers

File: PATENTS.md
Comment (robmoffat) [reply PATENTS.md:1]: added to README

File: CONTRIBUTING.md
Comment (kriswest) [inline CONTRIBUTING.md:111]:
```suggestion
 * Ensure all new source files include a header comment block containing the [Apache License v2.0 and your copyright information](http://www.apache.org/licenses/LICENSE-2.0#apply).
```

File: LICENSE
Comment (kriswest) [reply LICENSE:51]: @maoo I didn't understand this sentence:

>The link with Apache is when it says `If no source code license is included, the source code will be subject to the MIT License.` , and that's why we have added mentions to Apache in the README.

How does this reference to the MIT license link to the Apache license?? I thought defaulting to the Apache license was the main change we were making to the CSL and that this is the place that its done...

Comment (kriswest) [reply LICENSE:51]: @robmoffat unresolved this as I'm still not sure its right - or I'm just not understanding something

Comment (maoo) [reply LICENSE:51]: @kriswest - sorry, I admit that my explanation was not very clear/exhaustive ; I'll try again.

In order to keep things simple, we have decided to use CSL as-is, without any modification, for the following reasons:
1. It's a simpler approach, as we can point the FDC3 community to a known license, rather than a patched version of it - less docs, less questions, easier to integrate with other tools (ie SPDX, GitHub APIs, etc)
2. It can be used along with any other license, as long as it's stated in the project's README and/or Notices (MIT is just a fallback option)

Based on the second point, we made sure that the README.md (see section `License & Patents`) specifies that `Reference implementations and other software contained in FDC3 repositories is licensed under the Apache License, Version 2.0`.

We have validated this approach with our Legal team and we got their approval.

Comment (robmoffat) [reply LICENSE:51]: Also we updated all of the headers in source files to be Apache

**Comment by kriswest** [comment]
Understood - although it seems odd to have conflicting statements about licesing in the documentation.  The readme says 'software contained in FDC3 repositories is licensed under the [Apache License, Version 2.0](https://www.apache.org/licenses/LICENSE-2.0.html) unless otherwise noted' - this is such a note. Further, none of the schema files (of which there are many) have apache headers in them as neither JSON nor JSON schema support them, or the ability to add comments in the files to do so. Hence, its not entirely clear what license they are under due to the conflicting statements (it could be any of the 3 licenses TBH).

License & Patents section could deal with this more explicitly and with less conflict than it currently does.

**Comment by kriswest** [comment]
The FDC3 charter is another place that could provide a clarificaiton on schema licensing.

It matters as we are making the schema files the single source of truth for the standard definition for contexts and wire protocols - which leads me to think they should be explicitly under the CSL. Note that the 2.1 release also moves them out of the source folder and consolidates them all under a /schema folder.

**Comment by maoo** [comment]
Thanks for the feedback @kriswest , my answers below...

> it seems odd to have conflicting statements about licesing in the documentation. The readme says 'software contained in FDC3 repositories is licensed under the [Apache License, Version 2.0](https://www.apache.org/licenses/LICENSE-2.0.html) unless otherwise noted' - this is such a note.

I think we could/should remove the `unless otherwise noted`, since every code hosted in FDC3 will be released under Apache-2.0 , no exception. Other than this, do you see other conflicts in the proposed changes?

> None of the schema files (of which there are many) have apache headers in them as neither JSON nor JSON schema support them, or the ability to add comments in the files to do so. Hence, its not entirely clear what license they are under due to the conflicting statements (it could be any of the 3 licenses TBH).

I'd suggest to add a paragraph in `Notices.md` to clarify that `all files in the `/schema` folder are licenses under the Apache License, Version 2.0`; being code assets, I'd suggest to stick with Apache 2.0 in order to avoid confusion.

> The FDC3 charter is another place that could provide a clarificaiton on schema licensing.

Agreed, we can simply add the note above also in the charter page.

File: README.md
Comment (robmoffat) [inline README.md:154]:
```suggestion
- Reference implementations and other software contained in FDC3 repositories is licensed under the [Apache License, Version 2.0](https://www.apache.org/licenses/LICENSE-2.0.html). SPDX-License-Identifier: [Apache-2.0](https://spdx.org/licenses/Apache-2.0). 
```

Comment (robmoffat) [reply README.md:154]: remove "otherwise noted - we're using Apache.

File: docs/fdc3-charter.md
Comment (robmoffat) [inline docs/fdc3-charter.md:50]:
```suggestion
 - Versions of the FDC3 specification following 2.1 and subsequent draft specifications are licensed under the [Community Specification License 1.0](https://github.com/finos/FDC3/blob/45ca765e5ff9a44a1fa5437eb70cf876bf4898aa/LICENSE)
 - All code in the FDC3 specification (including JSON Schema) are licensed under [Apache 2.0](https://github.com/finos/FDC3/blob/master/LICENSE)
```

**Review by kriswest — APPROVED** [review]
LGTM

**Comment by kriswest** [comment]
@robmoffat this page needs updating to replace IP policy with CSL details: 
https://fdc3.finos.org/get-involved
The relevant file is here: https://github.com/finos/FDC3/blob/412b_CSL_License/website/data/get-involved/3.mdx

**Comment by robmoffat** [comment]
Missed this note - #1066

**Pull request merged**
